app/main/views/service_updates.py

```diff
@@ -54,33 +56,24 @@ def service_status_update_audits(day=None, page=1):
 @login_required
 @role_required('admin', 'admin-ccs-category')
 def service_update_audits():
-    form = ServiceUpdateAuditEventsForm(request.args, csrf_enabled=False)
-
-    if not form.validate():
-        return render_template(
-            "service_update_audits.html",
-            today=datetime.utcnow().strftime(DATETIME_FORMAT),
-            acknowledged=form.default_acknowledged(),
-            audit_events=[],
-            form=form
-        ), 400
-
-    audit_events = data_api_client.find_audit_events(
+    audit_events = list(data_api_client.find_audit_events_iter(
```

Comment:
I'm very worried about this approach. This will paginate though the API, downloading all unacknowledged audit events, potentially making hundreds or thousands of requests based on how many edits we get and how often admins check them.

It feels like this could easily end up in a situation were this page just times out and there's no way for us to recover without marking events as acknowledged in the API.

As far as I remember we've discussed this with @TheDoubleK and decided to use archived services table to drive this page.